### PR TITLE
Automate minification and generate source map

### LINKED ISSUE #31: Produce a minification map file
Source maps (generally `nameofscript.min.js.map`) are a standard that allow browsers to "unminify" the code for debugging in browser. 

http://www.html5rocks.com/en/tutorials/developertools/sourcemaps/

The map file is only downloaded upon developer action -- in Chrome's case, opening the developer console -- and so the map file is not routinely downloaded and shouldn't meaningfully increase costs. It will meaningfully increase the ability of the DAP team and its IT support staff to identify and solve problems that implementing agencies are experiencing.

A good minification toolset should be able to produce an accompanying source map file, which could then be referenced in the minified JavaScript and hosted next to it on `dap.digitalgov.gov`.

### PR DESCRIPTION
**Pull request by shawnbot**
This fixes #31 by introducing an [npm] configuration in `package.json` and adds a reusable script to compress the un-minified JS with [UglifyJS]. Here's how to use it:

1. Make sure that you have [Node.js] installed.
1. Run `npm install --dev` in the project directory.
1. Run `npm run minify` to regenerate the minified JS and corresponding source map.

This needs some documentation, preferably in the README. Also note that this will invalidate any PRs that touch the minified JS only, such as #20.

[Node.js]: https://nodejs.org/en/
[npm]: https://docs.npmjs.com/getting-started/what-is-npm
[UglifyJS]: https://github.com/mishoo/UglifyJS2

### CODE REVIEW

**Comment by Stargator** [comment]
@shawnbot I recommend that you include the change to the README as part of this PR so all the related changes are self-contained.

File: README.md [hunk 1]
```diff
@@ -79,3 +79,13 @@ This repository is maintained in its own GitHub organization, `digital-analytics
...
+
+## Minifying for production
+
+The production build of the script, `Universal-Federated-Analytics-Min.js`, is minified with [UglifyJS](https://github.com/mishoo/UglifyJS2). To build the production script:
+
+1. Clone this repository.
+1. Ensure that you have [Node.js](https://nodejs.org/en/) installed.
+1. Run `npm install --dev` in the project directory.
+1. Run `npm run minify` to regenerate the minified JS and corresponding [source map](https://developers.google.com/web/tools/chrome-devtools/javascript/source-maps).
+1. Commit changes to `Universal-Federated-Analytics-Min.js` and the `Universal-Federated-Analytics-Min.js.map` with any chances to the source script, `Universal-Federated-Analytics.js`.
```
Comment (Stargator) [inline README.md]: > any chances

Probably should be "any changes.

Otherwise looks good.

**Comment by Stargator** [comment]
@tdlowden Any thoughts on this PR?

**Comment by tdlowden** [comment]
Hi! Thanks. We have added sourcemap capabilities to v4.0. Would love thoughts! More in changelog:

https://github.com/digital-analytics-program/gov-wide-code/blob/master/changelog/changelog-4.0.md

**Pull request closed without merging**